### PR TITLE
Merge appsec repository

### PR DESCRIPTION
**Pull request by Anilm3**
### Description

The objective of this PR is to merge the appsec extension and helper process to this repository, including the ability to build and run unit tests. This PR will not focus on CI or packaging. For now, releases will be done using the releases in the DataDog/dd-appsec-php repository.

Work completed:
- Ported all source and tests over (unit + phpt, integrations will not be ported for now)
- Build & test extension through cmake
- Build tracer when required for testing the extension
- Build & test helper through cmake (not necessarily test, but tests are built through cmake)
- MPack dependency has been updated in the tracer (`ext/vendor/mpack`) and now also used by appsec
- Dependencies have been kept as a combination of `hunter`, git submodules, `FetchContent` and `ExternalProject`. However since all of these dependencies are from the helper process, no work needs to be done as it will eventually be merged with the sidecar.
- Clang tidy / format scripts have also been ported
- Support for generating coverage also ported
- Other files such as the license, contributing and readme have also been included without much change.

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

**Comment by realFlowControl** [comment]
Welcome on board 🚢 🎉 

image

File: Makefile [hunk 1]
```diff
@@ -153,6 +154,28 @@ test_extension_ci: $(SO_FILE) $(TEST_FILES) $(TEST_STUB_FILES)
 	$(RUN_TESTS_CMD) -d extension=$(SO_FILE) -m -s $$TEST_PHP_OUTPUT $(BUILD_DIR)/$(TESTS) && ! grep -e 'LEAKED TEST SUMMARY' $$TEST_PHP_OUTPUT; \
 	)
 
+build_appsec_ext: $(SO_FILE) clean_appsec_ext
```
Comment (bwoebi) [inline Makefile:157]: We can skip the _ext suffix, as this is the common operation I guess

Comment (Anilm3) [reply Makefile:157]: I'm not sure I'm keeping this in the Makefile but the objective was to build the extension and helper separately, although it's not clear to me now in which circumstance we'd use this Makefile when CMake would be faster, as it is properly incremental.

**Comment by bwoebi** [comment]
Superseded by #2267.

**Pull request closed without merging**